packages/studioCMS/src/defaultRoutes/components/BaseHead.astro

```diff
@@ -0,0 +1,87 @@
+---
+import type { z } from 'astro/zod';
+import Config from 'virtual:studiocms/config';
+import { headDefaults } from 'studiocms:components';
+import type { HeadConfigSchema } from '../../schemas/defaultFrontend';
+import { createHead } from '../../utils/head';
+import { fileWithBase } from '../../utils/base';
+import { extname } from 'node:path';
+
+// // Import the global.css file here so that it is included on
+// // all pages through the use of the <BaseHead /> component.
+import './global.css';
+
+const { defaultFrontEndConfig: { htmlDefaultHead, favicon } } = Config;
+
+interface Props {
+	title: string;
+	description: string;
+	lang?: string | undefined;
+	image?: string | undefined;
+}
+
+const canonicalURL = Astro.site ? new URL(Astro.url.pathname, Astro.site) : undefined;
+
+const faviconTypeMap = {
+	'.ico': 'image/x-icon',
+	'.gif': 'image/gif',
+	'.jpeg': 'image/jpeg',
+	'.jpg': 'image/jpeg',
+	'.png': 'image/png',
+	'.svg': 'image/svg+xml',
+};
+
+function isFaviconExt(ext: string): ext is keyof typeof faviconTypeMap {
+	return ext in faviconTypeMap;
+}
+
+const getFavicon = () => {
+	const ext = extname(favicon).toLowerCase();
+
+	if (!isFaviconExt(ext)) {
+		throw new Error(`Unsupported favicon type: ${ext}`);
+	}
+
+	const type = faviconTypeMap[ext];
+	return { href: fileWithBase(favicon), type };
+}
```

Comment:
***Nitpick***: The `type` logic here could be smaller and support more types using `mrmime` (library Astro uses) https://www.npmjs.com/package/mrmime

Reply:
This part was 100% stolen from starlight's Head Generator system :sweat_smile: i will look into the mrmime methods

Reply:
implemented